src/main/java/seedu/address/model/reminder/UniqueReminderList.java

```diff
@@ -0,0 +1,153 @@
+ * to be added or deleted to maintain the uniqueness of the ReminderList
+ * Supports a minimal set of list operations.
+ *
+ */
+public class UniqueReminderList implements Iterable<Reminder> {
+
+    private static UniqueReminderList reminderList;
+
+    private final ObservableList<Reminder> internalList;
+    private final ObservableList<Reminder> internalUnmodifiableList;
+    private final HashMap<Person, Reminder> personToReminderMap;
+
+    /**
+     * Constructor of UniqueReminderList
+     */
+    private UniqueReminderList() {
+        internalList = FXCollections.observableArrayList();
+        internalUnmodifiableList = FXCollections.unmodifiableObservableList(internalList);
+        personToReminderMap = new HashMap<>();
+    }
+
+    /**
+     * Returns true if the list contains no Reminders.
+     */
+    public boolean isEmpty() {
+        return internalList.isEmpty();
+    }
+
+    /**
+     * Returns true if the list contains an equivalent Reminder as the given argument.
+     */
+    public boolean contains(Reminder toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::equals);
+    }
+
+    /**
+     * Factory method of UniqueReminderList
+     *
+     * @return Produced UniqueReminderList object that is a Singleton.
+     */
+    public static UniqueReminderList getInstance() {
+        if (reminderList == null) {
+            reminderList = new UniqueReminderList();
+        }
+        return reminderList;
+    }
+
+    /**
+     * Adds a Reminder to the list.
+     * The Reminder must not already exist in the list.
+     */
+    public void add(Person person, Interaction interaction) {
+        requireAllNonNull(person, interaction);
+        //Person must have lead and interaciton to have a reminder
+        if (person.getLead() == null) {
+            return;
+        }
+        Reminder toAdd = new Reminder(person.getName(), person.getPhone(), person.getTags(),
+                person.getInteractions(), person.getLead(), interaction.getDate());
+        if (contains(toAdd)) {
+            throw new DuplicateReminderException();
+        }
+
+        if (personToReminderMap.containsKey(person)) {
+            internalList.remove(personToReminderMap.get(person));
+        }
+
+        internalList.add(toAdd);
+        personToReminderMap.put(person, toAdd);
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<Reminder> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    /**
+     * Returns the earliest reminder time in the {@code UniqueReminderList}
+     */
+    public long getEarliestReminderTime() {
+        if (internalList.isEmpty()) {
+            return -1;
+        }
+        return internalList.stream().mapToLong(Reminder::getDueTime).min().getAsLong();
```

Comment:
in relation to reminder manager above, if I'm understanding this correctly, the reminders are by dates right? So, there might be lots of reminders at the same day (e.g. multiple mins) here? I'm not sure how it's going to look like if that's the case